### ISSUE TITLE
CLI Configuration for Typegen

### ISSUE THREAD

**Issue by okj579** [sanity-io/sanity#7621]
**Is your feature request related to a problem? Please describe.**
Using Typegen across separate repositories (e.g. on for studio, one for the frontend) is difficult, as you can't mix local settings and those that are committed to the repository. The only configuration possibility at the moment is `sanity-typegen.json`.

**Describe the solution you'd like**
Configuration via CLI options and/or environment variables

**Describe alternatives you've considered**
I'm using an external script at the moment to gather the relevant options from constant, project and local settings, then resolve paths to be relative to the Sanity Studio root, write a temporary `sanity-typegen.json`, run typegen and clean up again afterwards. This is much more cumbersome than it has to be, though.

**Comment by sgulseth** [comment]
👋 We have had internal discussions around moving sanity-typegen.json config into the `sanity.config.ts`, as a separate object. The config would look something similar to:

```ts
import {defineConfig} from 'sanity'

export default defineConfig({
  ... // the rest of your config
  schemaExtract: {
     ...
  },
  typegen: {
    generates: 'sanity.types.ts` // or process.env.TYPEGEN_OUTPUT || 'sanity.types.ts`
  },
})
```

Since this all being executed then during runtime it should work by reading options from the environment. 

Would this solve your use case?

**Comment by okj579** [comment]
Yes, I think with that I could set all relevant settings on the command line without temporary files and the like.

**Comment by kbrabrand** [comment]
👋 @okj579 Support for configuring typegen through the `sanity.cli.ts` is available since [`v4.19.0`](https://github.com/sanity-io/sanity/releases/tag/v4.19.0).
